client/pages/upload.js

```diff
@@ -234,19 +235,59 @@ const Upload = ({ token, userId }) => {
       setDropError(e.message);
     }
   }, []);
+  const onPosterDrop = useCallback((acceptedFiles) => {
+    try {
+      const [file] = acceptedFiles;
+      if (file) {
+        const reader = new FileReader();
+        reader.onload = async () => {
+          console.log(
+            `[DEBUG] Poster upload complete: ${reader.result.slice(0, 64)}...`
+          );
+          const [, posterB64] = reader.result.split("base64,");
+          setPosterFile({ name: file.name, b64: posterB64 });
+        };
+        reader.onerror = () => {
+          console.log(`[DEBUG] Poster upload error: ${reader.error}`);
+          // Gérer les erreurs, si nécessaire
+        };
+        reader.readAsDataURL(file);
+      }
+    } catch (e) {
+      console.error(e);
+      // Gérer les erreurs, si nécessaire
+    }
+  }, []);
 
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     onDrop,
     accept: { "application/x-bittorrent": [".torrent"] },
     maxFiles: 1,
   });
+  const {
+    getRootProps: getPosterRootProps,
+    getInputProps: getPosterInputProps,
+    isDragActive: isPosterDragActive,
+  } = useDropzone({
+    onDrop: onPosterDrop,
+    accept: {
+      "image/jpeg": [".jpg", ".jpeg"],
+      "image/png": [".png"],
+    },
+    maxFiles: 1,
+  });
+  function isPngImage(data) {
+    const pngHeader = "data:image/png;base64,";
+    return data.startsWith(pngHeader);
+  }
 
   const handleUpload = async (e) => {
     e.preventDefault();
     setLoading(true);
     const form = new FormData(e.target);
 
     try {
+      form.append("poster", posterFile.b64); // Ajoutez la valeur de l'image de l'affiche à la requête
```

Comment:
Does this need to be appended to the form when you are submitting it below in the fetch request?